### PR TITLE
Stupid ui router

### PR DESCRIPTION
**Pull request by jamesgorrie**
Slashes keep breaking - now they look right, but are double encoded when read from the `stateParams`.
[Following the following advice](https://github.com/angular-ui/ui-router/issues/1119#issuecomment-134603427), this seems to work in all cases.

Sorry @akash1810 - I think this is what you did the first time.

### CODE REVIEW

File: kahuna/public/js/search/index.js [hunk 1]
```diff
@@ -24,14 +24,26 @@ export var search = angular.module('kahuna.search', [
     'data-structure.list-factory',
     'data-structure.ordered-set-factory',
     'gr.topBar',
-    'grPanel'
+    'grPanel',
+    'ui.router'
```
Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: This was added as it should be here.

File: kahuna/public/js/search/index.js [hunk 2]
```diff
@@ -24,14 +24,26 @@ export var search = angular.module('kahuna.search', [
...
-               function($stateProvider) {
+search.config(['$stateProvider', '$urlMatcherFactoryProvider',
+               function($stateProvider, $urlMatcherFactoryProvider) {
+
+    // Query params slashes are encoded that mess up our seach query when we search for e.g. `24/7`.
+    // see: https://github.com/angular-ui/ui-router/issues/1119#issuecomment-64696060
+    // TODO: Fix this
+    $urlMatcherFactoryProvider.type('Slashed', {
+        encode: val => val || '',
+        decode: val => val || '',
```
Comment (akash1810) [inline kahuna/public/js/search/index.js]: Will this decode `%2F` to `%2F` rather than `/`?

**Comment by tsop14** [comment]
:+1: :tophat:

**Comment by tsop14** [comment]
Happy to test this in TEST with the cases I saw.

File: kahuna/public/js/search/index.js (hunk 2 above)
Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: If you search for %2F, you will get %2F.
e.g.
* create label: %2F
* search for #%2F

input value: #%2F
url value: %23%252F

Comment (jamesgorrie) [inline kahuna/public/js/search/index.js]: Saying that, I don't want to return a string for undefineds, so I have updated it.

**Pull request merged**